programs/drift/src/controller/orders.rs

```diff
@@ -3258,6 +3258,20 @@ pub fn fill_spot_order(
         )?
     }
 
+    {
+        let spot_market = spot_market_map.get_ref(&user.orders[order_index].market_index)?;
```

Comment:
if order isn't open, this will always be 0

Reply:
updated to use `order_market_index` instead at top of function, see any issues w/ that?